### PR TITLE
[All] Rename SOFA_NO_OPENGL + Relocate Sofa.GL + Improve sofa_add_* macros

### PR DESCRIPTION
**Pull request by guparan**
1. The CMake option to enable/disable OpenGL is now `SOFA_WITH_OPENGL`.
2. SOFA_NO_OPENGL is completely removed from SOFA sources (only the CMake variable persists).
3. Sofa.GL is added by SofaKernel/modules/CMakeLists.txt (not by SofaFramework/CMakeLists.txt anymore).
4. sofa_add_* macros are now capable of creating dependent options. 
Example:   
```cmake
sofa_add_module(Sofa.GL Sofa.GL ON # the 3 usual sofa_add_* arguments
    WHEN_TO_SHOW "SOFA_WITH_OPENGL" # [new arg] if SOFA_WITH_OPENGL is true, then the option is showed in CMake-GUI
    VALUE_IF_HIDDEN OFF # [new arg] if SOFA_WITH_OPENGL is false, then the option is hidden and set to OFF
    )
```
5. SofaComponentAll does not fail anymore at configure time if some modules are not found. It will cover as many modules as possible. **This will fix our nightly minimal builds.**
______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: applications/plugins/Compliant/mapping/AdditionMapping.h [hunk 1]
```diff
@@ -121,7 +121,7 @@ class SOFA_Compliant_API AdditionMapping : public ConstantAssembledMapping<TIn,
     void draw(const core::visual::VisualParams* vparams) override
     {
 
-#ifndef SOFA_NO_OPENGL
+#ifdef COMPLIANT_HAVE_SOFA_GL
```
Comment (fredroy) [inline applications/plugins/Compliant/mapping/AdditionMapping.h]: hummmm, should not it be #if COMPLIANT_HAVE_SOFA_GL ==1 ? 🤠

Comment (fredroy) [reply applications/plugins/Compliant/mapping/AdditionMapping.h]: (and same for the others)

File: applications/projects/SofaGuiGlut/CMakeLists.txt [hunk 2]
```diff
@@ -14,8 +15,6 @@ set(SOURCE_FILES
 
 list(APPEND SOURCE_FILES Main.cpp)
 
-find_package(GLUT REQUIRED)
```
Comment (fredroy) [inline applications/projects/SofaGuiGlut/CMakeLists.txt:17]: GLUT and Sofa.GL are not related so you must keep that

**Review by fredroy — CHANGES_REQUESTED** [review]

**Comment by fredroy** [comment]
and did you try with out-of-tree compilation with SofaPython3 for example ? (as the CI does not test this mode for the moment)

File: applications/plugins/Compliant/mapping/AdditionMapping.h (hunk 1 above)
Comment (guparan) [reply applications/plugins/Compliant/mapping/AdditionMapping.h]: Haha you god damn right

Comment (guparan) [reply applications/plugins/Compliant/mapping/AdditionMapping.h]: Is the `== 1` really needed though?

File: applications/projects/SofaGuiGlut/CMakeLists.txt (hunk 2 above)
Comment (guparan) [reply applications/projects/SofaGuiGlut/CMakeLists.txt:17]: Actually it was duplicated. See line 5 :arrow_up:

Comment (fredroy) [reply applications/projects/SofaGuiGlut/CMakeLists.txt:17]: ah ok 👍

File: applications/plugins/Compliant/mapping/AdditionMapping.h (hunk 1 above)
Comment (fredroy) [reply applications/plugins/Compliant/mapping/AdditionMapping.h]: dont think so actually, just the if should be enough indeed

**Review by fredroy — APPROVED** [review]

**Comment by guparan** [comment]
[ci-build]

**Pull request merged**